File: app/src/main/java/com/example/l091735/weather_modified_app/dependency/injection/modules/LocationModule.java
package com.example.l091735.weather_modified_app.dependency.injection.modules;

import android.app.Application;

import com.example.l091735.weather_modified_app.utils.Codes;
import com.google.android.gms.common.api.GoogleApiClient;
import com.google.android.gms.location.LocationRequest;
import com.google.android.gms.location.LocationServices;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by L091735 on 28/10/2016.
 */

@Module
public class LocationModule {

    private Application application;

    public LocationModule(Application application) {
        this.application = application;
    }

    @Provides
    @Singleton
    LocationRequest getLocationRequestInstance() {
        LocationRequest locationRequest = new LocationRequest();
        locationRequest.setPriority(LocationRequest.PRIORITY_BALANCED_POWER_ACCURACY);
        locationRequest.setInterval(Codes.LOCATION_INTERVAL_TIME);
        locationRequest.setFastestInterval(Codes.LOCATION_FASTEST_INTERVAL_TIME);

        return locationRequest;
    }

    @Provides
    @Singleton
    GoogleApiClient.Builder getGoogleApiClientBuilderInstance() {

        return new GoogleApiClient.Builder(application).addApi(LocationServices.API);
    }
}

File: app/src/main/java/com/westpac/l091735/weatherforecastapp/di/modules/LocationModule.kt
package com.westpac.l091735.weatherforecastapp.di.modules

import android.app.Application
import com.google.android.gms.common.api.GoogleApiClient
import com.google.android.gms.location.LocationRequest
import com.google.android.gms.location.LocationServices
import com.westpac.l091735.weatherforecastapp.utils.Codes
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class LocationModule {


    @Provides
    @Singleton
    internal fun providesLocationRequestInstance(): LocationRequest {
        val locationRequest = LocationRequest()
        locationRequest.priority = LocationRequest.PRIORITY_BALANCED_POWER_ACCURACY
        locationRequest.interval = Codes.LOCATION_INTERVAL_TIME
        locationRequest.fastestInterval = Codes.LOCATION_FASTEST_INTERVAL_TIME

        return locationRequest
    }

    @Provides
    @Singleton
    internal fun providesGoogleApiClientBuilderInstance(application: Application): GoogleApiClient.Builder {
        return GoogleApiClient.Builder(application).addApi(LocationServices.API)
    }
}